src/Filter/FilterComposite.php

```diff
@@ -124,46 +124,43 @@ public function removeFilter(string $name) : void
      */
     public function filter(string $property) : bool
     {
-        $andCount = count($this->andFilter);
-        $orCount = count($this->orFilter);
-        // return true if no filters are registered
-        if ($orCount === 0 && $andCount === 0) {
+        return $this->atLeastOneOrFilterIsTrue($property) && $this->allAndFiltersAreTrue($property);
+    }
+
+    private function atLeastOneOrFilterIsTrue(string $property)
```

Comment:
the method name is a bit confusing, because it returns also true when there is no any `orFilters`.

also: we can add return type hint on the method: `: bool`

the same comment for other function (for `andFilters`).

Reply:
What would be your suggestion for the name of the method ?

Reply:
That's the problem.... ;) @froschdesign maybe any suggestion here?